### ISSUE TITLE
No attribute 'exponential_query_iterator' when running descriptor_eval.py 

### ISSUE THREAD

**Issue by courao** [facebookresearch/vsc2022#10]
Hi, when I reproduce the baseline, I meet this problem.

Here is detail error message
![image](https://user-images.githubusercontent.com/12403083/220813190-7a3d722e-178a-4b76-8b36-2bff86232c79.png)

AttributeError: module 'faiss.contrib.exhaustive_search' has no attribute 'exponential_query_iterator'

It seems faiss contrib remove the attribute 'exhaustive_search.exponential_query_iterator'? Which version of faiss should I use?

**Comment by courao** [comment]
[Update] faiss-cpu version seems no such problem.

**Comment by edpizzi** [comment]
We tested on version 1.7.2 and the main branch near when 1.7.3 was released, so we expect 1.7.3 to work as well. If the error was for an earlier version of faiss, that is expected.

The faiss exhaustive_search library was changed significantly in recent faiss versions, making it difficult to implement range search based on a global result budget in a way that works for both 1.7.2 and earlier faiss versions (at least for inner-product similarity).

For a version that works with earlier faiss versions, this codebase's `_global_threshold_knn_search` is similar to `search_with_capped_res` from the Image Similarity Challenge codebase, but the ISC version targets earlier versions of faiss:

https://github.com/facebookresearch/isc2021/blob/main/isc/descriptor_matching.py#L63
